cylc/flow/subprocpool.py

```diff
@@ -69,29 +69,39 @@ def _killpg(proc, signal):
 def get_func(func_name, src_dir):
     """Find and return an xtrigger function from a module of the same name.
 
-    Can be in <src_dir>/lib/python, CYLC_MOD_LOC, or in Python path.
+    Can be in <src_dir>/lib/python, in Python path, or defined via an
+    entry_point. Loctions checked are in this order.
+
     Workflow source directory passed in as this is executed in an independent
     process in the command pool and therefore doesn't know about the workflow.
 
     """
     if func_name in _XTRIG_FUNCS:
         return _XTRIG_FUNCS[func_name]
+
     # First look in <src-dir>/lib/python.
     sys.path.insert(0, os.path.join(src_dir, 'lib', 'python'))
     mod_name = func_name
     try:
         mod_by_name = __import__(mod_name, fromlist=[mod_name])
     except ImportError:
-        # Then look in built-in xtriggers.
-        mod_name = "%s.%s" % ("cylc.flow.xtriggers", func_name)
-        try:
-            mod_by_name = __import__(mod_name, fromlist=[mod_name])
-        except ImportError:
-            raise
+        # Look for xtriggers via entry_points for external sources.
+        # Do this after the lib/python and PYTHONPATH approaches to allow
+        # users to override entry_point definitions with local/custom
+        # implementations.
+        for entry_point in iter_entry_points('cylc.xtriggers'):
+            entries = entry_point.resolve()()
+            if func_name in entries:
+                _XTRIG_FUNCS[func_name] = entries[func_name]
+                return _XTRIG_FUNCS[func_name]
```

Comment:
This is importing all triggers, whether the workflow requested them or not.

Suggest registering xtriggers via entry points (rather than registering xtrigger packages) and filtering by name.

```suggestion
        for entry_point in iter_entry_points('cylc.xtriggers'):
            if entry_point.name == func_name:
                _XTRIG_FUNCS[func_name] = entry_point.resolve()
                return _XTRIG_FUNCS[func_name]
```

Reply:
Yes, much better and cleaner. Thanks for your help and advice.